src/services/ProjectTokenHistoricalPriceService.ts

```diff
@@ -159,31 +159,31 @@ export class ProjectTokenHistoricalPriceService {
 
 
       if (todaysSwaps?.length) {
-        const priceAtTimePoint = this.nearestUSDPriceAtTimestamp(prices, todaysSwaps[todaysSwaps.length-1].timestamp );
+        const usdPriceAtTimePoint = this.nearestUSDPriceAtTimestamp(prices, todaysSwaps[todaysSwaps.length-1].timestamp );
         returnArray.push({
           time: timestamp,
           price: (
-            (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountIn)) /
+            (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountIn) * (1 + lbpMgr.swapFeePercentage)) /
             (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountOut)) *
-            priceAtTimePoint
+            usdPriceAtTimePoint
           ),
         });
 
         previousTimePoint = (
-          (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountIn)) /
+          (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountIn) * (1 + lbpMgr.swapFeePercentage)) /
```

Comment:
Just to make sure: the swap fee is not already included in the swap data that we receive from the subgraph per swap?

Reply:
correct, we confirmed this looking at the contracts.  wouldn't hurt if you want to triple check @fabianschu .